Source/Bogus/DataSets/Date.cs

```diff
@@ -161,14 +153,19 @@ public DateTime Between(DateTime start, DateTime end)
       /// <param name="end">End time</param>
       public DateTimeOffset BetweenOffset(DateTimeOffset start, DateTimeOffset end)
       {
-         var minTicks = Math.Min(start.Ticks, end.Ticks);
-         var maxTicks = Math.Max(start.Ticks, end.Ticks);
+         var startTicks = start.ToUniversalTime().Ticks;
+         var endTicks = end.ToUniversalTime().Ticks;
+
+         var minTicks = Math.Min(startTicks, endTicks);
+         var maxTicks = Math.Max(startTicks, endTicks);
 
          var totalTimeSpanTicks = maxTicks - minTicks;
 
          var partTimeSpan = RandomTimeSpanFromTicks(totalTimeSpanTicks);
 
-         return new DateTimeOffset(minTicks, start.Offset) + partTimeSpan;
+         var dateTime = new DateTime(minTicks, DateTimeKind.Unspecified) + partTimeSpan;
+
+         return new DateTimeOffset(dateTime + start.Offset, start.Offset);
```

Comment:
I'm going to replace these two lines with:

```csharp
         if( start < end )
            return start + partTimeSpan;
         else
            return end + partTimeSpan;
```

Given an invalid `start` time; the function will still return an invalid `DateTimeOffset`.

![image](https://user-images.githubusercontent.com/478118/111921243-c5792600-8a50-11eb-9c4a-d58b9120e257.png)

With the code change, at least the method returns a valid time, albeit outside the "specified" time range; which is fine I guess, because as it stands now, it would still have been outside the "2:59 AM - 3:00 AM" time as noted above.

![image](https://user-images.githubusercontent.com/478118/111921287-10933900-8a51-11eb-859c-bc79bf1765c6.png)

Reply:
That seems legit :-)

Reply:
I guess to my mind, if the caller supplies impossible DateTime values to begin with, all bets are off. My specific use case doesn't intersect with that, so I don't have strong opinions about how exactly it behaves in that edge case.

Reply:
Thinking it through further, if one wanted to be _really_ thorough, then the algorithm would need to include some sort of "compute _real_ DateTime range" functionality, so that `start` and `end` were adjusted if they lay inside an impossible time span, and this would then also throw if `start` and `end` both lay within the same transition window (so that no real date/time `x` would satisfy `start <= x <= end`). This would be in addition to generating the actual bogus value in UTC, to avoid transition windows in the middle of the range. I think this would be the most polished form of this, but I'm not sure if it's worth the effort.

Reply:
Pseudocode:
```
void ComputeRealRange(ref DateTime start, ref DateTime end)
{
  if (start > end)
    Swap(ref start, ref end);

  var (windowStart, windowEnd) = GetDSTTransitionWindow(start.Year);

  if (windowStart <= start <= windowEnd)
    start = windowEnd;

  (windowStart, windowEnd) = GetDSTTransitionWindow(end.Year);

  if (windowStart <= end <= windowEnd)
    end = windowStart;

  if (start > end)
    throw new Exception("Impossible");
}
```

Reply:
With this approach, a supplied range as shown where `start` lay within the impossible range and `end` was exactly the first next valid `DateTime` value, it would always return exactly the `end` value because the _effective_ `start` would be bumped to the end of the transition window and would coincide with `end`.